modules/navier_stokes/src/actions/NSFVAction.C

```diff
@@ -2571,13 +2580,15 @@ NSFVAction::checkPassiveScalarParameterErrors()
 
 void
 NSFVAction::checkDependentParameterError(const std::string main_parameter,
-                                         const std::vector<std::string> dependent_parameters)
+                                         const std::vector<std::string> dependent_parameters,
+                                         const bool should_be_defined)
 {
   for (const auto & param : dependent_parameters)
-    if (_pars.isParamSetByUser(param))
+    if (_pars.isParamSetByUser(param) == !should_be_defined)
       paramError(param,
-                 "This parameter should not be given by the user with the corresponding " +
-                     main_parameter + " setting!");
+                 "This parameter should " + std::string(should_be_defined ? "" : "not") +
```

Comment:
Did the compiler require explicitly typing this?

Reply:
Yes, conversion from char[13] caused compile-time-errors.